internal/repository/indexmap.go

```diff
@@ -34,7 +34,7 @@ const (
 )
 
 // add inserts an indexEntry for the given arguments into the map,
-// using id as the key.
+// does not insert duplicates (all values identical to an existing one).
```

Comment:
This is a new sentence and it contradicts the comment on type indexMap.

```
 // add inserts an indexEntry for the given arguments into the map.
// It allows duplicate ids, but not fully duplicate entries.
```

The comment on indexMap needs an update too.